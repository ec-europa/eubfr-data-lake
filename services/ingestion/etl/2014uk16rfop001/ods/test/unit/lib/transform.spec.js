/**
 * @jest-environment node
 */

import mapper from '../../../src/lib/transform';
import testRecord from '../../stubs/record.json';

describe('2014uk16rfop001 ODS transformer', () => {
  let result = {};

  beforeAll(() => {
    result = mapper(testRecord);
  });

  test('Returns null when record is not provided', () => {
    expect(mapper()).toBe(null);
  });

  test('Produces correct JSON output structure', () => {
    expect(result).toMatchSnapshot();
  });
});
