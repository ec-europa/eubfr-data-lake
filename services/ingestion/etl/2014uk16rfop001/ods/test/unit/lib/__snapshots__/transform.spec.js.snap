// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2014uk16rfop001 ODS transformer Produces correct JSON output structure 1`] = `
Object {
  "action": "",
  "budget": Object {
    "eu_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "funding_area": Array [],
    "mmf_heading": "",
    "other_contrib": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "private_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "public_fund": Object {
      "currency": "",
      "raw": "",
      "value": 0,
    },
    "total_cost": Object {
      "currency": "GBP",
      "raw": 3636525,
      "value": 3636525,
    },
  },
  "call_year": "",
  "complete": false,
  "description": "Beneficiary identifier (E-claims ref): 05R16P00351 
Sector NACE group level: M71.20 Technical testing and analysis 
SANI reference of the aid measure: SA 39161 
Objective of the aid: Fundamental research (Art. 25(2)(a)) 
",
  "ec_priorities": Array [],
  "media": Array [],
  "programme_name": "",
  "project_id": "3d3de125a4aa564b78a2979649ed587a",
  "project_locations": Array [
    Object {
      "address": "",
      "centroid": null,
      "country_code": "GB",
      "location": null,
      "nuts": Array [
        Object {
          "code": "Cornwall and Isles of Scilly",
          "level": 2,
          "name": "",
          "year": null,
        },
      ],
      "postal_code": "",
      "region": "",
      "town": "",
    },
  ],
  "project_website": "",
  "related_links": Array [],
  "reporting_organisation": "Member states",
  "results": Object {
    "available": "",
    "result": "",
  },
  "status": "",
  "sub_programme_name": "",
  "success_story": "",
  "themes": Array [],
  "third_parties": Array [
    Object {
      "address": "",
      "country": "GB",
      "email": "",
      "name": "CETO Wave Energy UK Ltd",
      "phone": "",
      "region": "",
      "role": "Beneficiary",
      "type": "",
      "website": "",
    },
  ],
  "timeframe": Object {
    "from": "2016-11-21T22:00:00.000Z",
    "from_precision": "day",
    "to": null,
    "to_precision": "day",
  },
  "title": "European Regional Development Fund State Aid",
  "type": Array [],
}
`;
